### PR TITLE
First pass at CONTRIBUTING

### PR DESCRIPTION
**Pull request by jkarni**
Comments and suggestions very welcome!

### CODE REVIEW

**Comment by jkarni** [comment]
@aaronlevin @soenkehahn some of the ideas we discussed today.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+we decide that it can, you can still keep maintainership over it.
+
+Whether or not you want your package to be in the repo, create an issue with
+the `news` label if you make a new package so we can know about it!
+
+## Release policy
+
+We are currently moving to a more aggresive release policy, so that you can get
+what you contribute from Hackage fairly soon. However, note that prior to major
+releases it may take some time in between releases. If you think you're change
```
Comment (alpmestan) [inline CONTRIBUTING.md]: `you're` -> `your`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+is small enough that you should be backported to released major versions, say
```
Comment (alpmestan) [inline CONTRIBUTING.md]: that **it** should be backported. don't backport our contributors! =)

**Comment by alpmestan** [comment]
Aside from the minor comments, LGTM.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+Whether or not you want your package to be in the repo, create an issue with
+the `news` label if you make a new package so we can know about it!
+
+## Release policy
+
+We are currently moving to a more aggresive release policy, so that you can get
+what you contribute from Hackage fairly soon. However, note that prior to major
+releases it may take some time in between releases. If you think you're change
+is small enough that it should be backported to released major versions, say
+so in the issue or PR.
```
Comment (soenkehahn) [inline CONTRIBUTING.md]: Personally I would omit the comment about backporting. We can talk about that when/if someone wants it.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+one of the most powerful ways of using `servant`, and a wonderful way of
+getting to know it better. If you do write a new combinator, we would love to
+know about it! Either hop on #servant on freenode and let us know, or open an
+issue with the `news` tag (which we will close when we read it).
+
+As for adding them to the main repo: maintaining combinators can be expensive,
+since official combinators must have instances for all classes (and new classes
+come along fairly frequently). We therefore have to be quite selective about
+those that we accept. If your considering writing a new combinator, open an
+issue to discuss it first!
```
Comment (soenkehahn) [inline CONTRIBUTING.md]: Maybe we should add:
> (You could always release your combinator as a separate package, of course.)

**Comment by soenkehahn** [comment]
+1

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+We encourage people to experiment with new combinators and instances - it is
+one of the most powerful ways of using `servant`, and a wonderful way of
+getting to know it better. If you do write a new combinator, we would love to
+know about it! Either hop on #servant on freenode and let us know, or open an
+issue with the `news` tag (which we will close when we read it).
+
+As for adding them to the main repo: maintaining combinators can be expensive,
+since official combinators must have instances for all classes (and new classes
+come along fairly frequently). We therefore have to be quite selective about
+those that we accept. If your considering writing a new combinator, open an
```
Comment (imalsogreg) [inline CONTRIBUTING.md]: your -> you're

**Pull request merged**